api/notd/token_manager.py

```diff
@@ -435,6 +441,72 @@ async def update_activity_for_collection(self, address: str, startDate: datetime
                 else:
                     await self.saver.create_collection_hourly_activity(connection=connection, address=retrievedCollectionActivity.address, date=retrievedCollectionActivity.date, transferCount=retrievedCollectionActivity.transferCount, saleCount=retrievedCollectionActivity.saleCount, totalValue=retrievedCollectionActivity.totalValue, minimumValue=retrievedCollectionActivity.minimumValue, maximumValue=retrievedCollectionActivity.maximumValue, averageValue=retrievedCollectionActivity.averageValue,)
 
+
+    async def update_token_attributes_for_all_collections_deferred(self) -> None:
+        await self.tokenQueue.send_message(message=UpdateTokenAttributesForAllCollectionsMessageContent().to_message())
+
+    async def update_token_attributes_for_all_collections(self) -> None:
+        for _, address in enumerate(GALLERY_COLLECTIONS):
+            await self.update_token_attributes_for_collection(address=address)
+
+    async def update_token_attributes_for_collection(self, address: str) -> None:
+        startDate = date_util.datetime_from_now()
+        latestUpdate = await self.retriever.get_latest_update_by_key_name(key='token_attributes')
+        latestProcessedDate = latestUpdate.date
+        logging.info(f'Finding changed tokens since {latestProcessedDate}')
+        updatedTokenMetadatasQuery = (
+            TokenMetadatasTable.select()
+            .with_only_columns([TokenMetadatasTable.c.registryAddress, TokenMetadatasTable.c.tokenId])
+            .where(TokenMetadatasTable.c.registryAddress == address)
+            .where(TokenMetadatasTable.c.updatedDate >= latestProcessedDate)
+        )
+        updatedTokenMetadatasQueryResult = await self.retriever.database.execute(query=updatedTokenMetadatasQuery)
+        updatedTokenMetadatas = set(updatedTokenMetadatasQueryResult)
+        logging.info(f'Scheduling processing for {len(updatedTokenMetadatas)} tokens in collection {address}')
+        messages = [UpdateCollectionTokenAttributesMessageContent(registryAddress=registryAddress, tokenId=tokenId).to_message() for (registryAddress, tokenId) in updatedTokenMetadatas]
+        await self.tokenQueue.send_messages(messages=messages)
+        await self.saver.update_latest_update(latestUpdateId=latestUpdate.latestUpdateId, date=startDate)
+
+    async def update_collection_token_attributes_deferred(self, registryAddress: str, tokenId: str) -> None:
+        await self.tokenQueue.send_messages(messages=UpdateCollectionTokenAttributesMessageContent(registryAddress=registryAddress, tokenId=tokenId).to_message())
+
+    async def update_collection_token_attributes(self, registryAddress: str, tokenId: str) -> None:
+        retrievedTokenAttributes: List[RetrievedTokenAttribute] = await self.tokenAttributeProcessor.get_token_attributes(registryAddress=registryAddress, tokenId=tokenId)
```

Comment:
why do you need to explicity typing on this line? it should be coming from the typing of the function